From 7b857175d21c730a05915733421ff7ff78dc1c24 Mon Sep 17 00:00:00 2001
From: David Mc Lean <david.mclean@oracle.com>
Date: Fri, 9 Aug 2019 12:47:13 -0700
Subject: [PATCH 18/24] dtrace: add rcu_irq_exit and rcu_nmi_exit_common to FBT
 blacklist

Additional FBT blacklist entry needed.  Without this change, systems under
test were seen to fail at test/stress/dtrace-util/tst.DestructWithModule.sh
and test/stress/fbtsafety/ tests.

rcu_irq_exit is being added to the blacklist because it inlines
rcu_nmi_exit_common which is unsafe for FBT.

The symptom seen has been a system hang, requiring a hard reboot.
This failure was seen to be easily repeatable.

rcu_nmi_exit_common is being added also, just in case it is ever used
without being inlined.

Signed-off-by: David  Mc Lean <david.mclean@oracle.com>
Reviewed-by: Kris Van Hees <kris.van.hees@oracle.com>
---
 arch/x86/kernel/fbt_blacklist.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/fbt_blacklist.h b/arch/x86/kernel/fbt_blacklist.h
index fd599859b47e..1663b0b1be28 100644
--- a/arch/x86/kernel/fbt_blacklist.h
+++ b/arch/x86/kernel/fbt_blacklist.h
@@ -57,6 +57,8 @@ BL_DENTRY(void *, rcu_dynticks_curr_cpu_in_eqs)
 BL_DENTRY(void *, rcu_dynticks_eqs_exit)
 BL_DENTRY(void *, trace_rcu_dyntick)
 BL_DENTRY(void *, rcu_nmi_exit)
+BL_DENTRY(void *, rcu_irq_exit)
+BL_DENTRY(void *, rcu_nmi_exit_common)
 BL_DENTRY(void *, rcu_dynticks_eqs_enter)
 BL_DENTRY(void *, ist_exit)
 
-- 
2.28.0

